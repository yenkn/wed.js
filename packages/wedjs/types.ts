export type PromiseReturnType<T> = T extends (...args: any[]) => Promise<infer P> ? P : any
