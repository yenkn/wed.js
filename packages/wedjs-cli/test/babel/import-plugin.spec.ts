import * as babel from '@babel/core'
import importRequestPlugin from '../../build/babel/plugin/lambda-import-to-request'
import path from 'path'

const code = `
import types, { getUserList, createUser } from './lambda/test'

const useUserList = () => {
  const fetch = getUserList()
  
  useEffect(() => {
    fetch()
  }, [])
}

const App1 = () => {
  const { data } = useRequest(() => getUserList())
  const { data1 } = useRequest(() => types['getUserList']())
  const { data2 } = useRequest(getUserList, [keyword])
  const { data3 } = useRequest((page) => getUserList('keyword', page), [page])
  
  const handleClick = useCallback(() => {
  	const fetch = () => createUser(state)
    fetch()
  }, [state])
  
  return <div><Dash/></div>
}

function App2(props) {
  const { data } = useUserList()
}

export function App3() {
}

export default function() {
}

export const App4 = () => {
}

const App5 = () => <div></div>
`


describe('import-plugin-test', () => {
  beforeAll(() => {
    process.chdir(__dirname)
  })

  it('plugin-works', async () => {
    const options = {
      presets: [
        ["@babel/preset-react"],
      ],
      plugins: [
        [importRequestPlugin, { lambdaPath: './lambda' }]
      ]
    }

    const res = await babel.transformAsync(code, {
      ...options,
      filename: path.resolve('./App.tsx'),
    })
    // console.log(res.code)
    expect(res.code).toMatchSnapshot()
  })
})
