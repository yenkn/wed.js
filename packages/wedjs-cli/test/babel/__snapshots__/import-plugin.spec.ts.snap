// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import-plugin-test plugin-works 1`] = `
"import { requestLambda as __requestLambda, useLambdaContext as __useLambdaContext } from 'wedjs/cjs/lambda/client';
const __wed_lambda_config = {};

const useUserList = () => {
  const __wed_lambda_config = __useLambdaContext();

  const fetch = __requestLambda(\\"test/getUserList\\", __wed_lambda_config);

  useEffect(() => {
    fetch();
  }, []);
};

const App1 = () => {
  const __wed_lambda_config = __useLambdaContext();

  const {
    data
  } = useRequest(() => __requestLambda(\\"test/getUserList\\", __wed_lambda_config));
  const {
    data1
  } = useRequest(() => __requestLambda(\\"test\\", __wed_lambda_config));
  const handleClick = useCallback(() => {
    const fetch = () => __requestLambda(\\"test/createUser\\", __wed_lambda_config, state);

    fetch();
  }, [state]);
  return React.createElement(\\"div\\", null, React.createElement(Dash, null));
};

function App2(props) {
  const __wed_lambda_config = __useLambdaContext();

  const {
    data
  } = useUserList();
}

export function App3() {
  const __wed_lambda_config = __useLambdaContext();
}
export default function () {
  const __wed_lambda_config = __useLambdaContext();
}
export const App4 = () => {
  const __wed_lambda_config = __useLambdaContext();
};

const App5 = () => {
  const __wed_lambda_config = __useLambdaContext();

  return React.createElement(\\"div\\", null);
};"
`;
